services/AlertsManagement/actionRules/Deploy-AlertProcessingRule-Deploy.json

```diff
@@ -358,6 +400,32 @@
                                 }
                               ]
                             }
+                          },
+                          {
```

Comment:
This duplication of code, I think you can avoid it by simplifying the condition, can you check? A condition like:

`"condition": "[[empty(parameters('BYOAlertProcessingRule'))]",`

And use an If statement to determine the value, should be something like this:

`"actionGroupIds": "[[if(not(empty(parameters('BYOActionGroup'))),variables('varAGIds'),concat(subscription().Id, '/resourceGroups/', parameters('ALZMonitorResourceGroupName'), '/providers/microsoft.insights/actionGroups/', 'ag-AMBA-', subscription().displayName, '-001'))]",`

Reply:
Good spot 👍 ... I am testing it out but it looks like the syntax does not work giving a bad payload error. from the dev code point of view it is correct, compiles and deploys but the remediation task fails with this error: 

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/26008435/e7dc06b2-149e-4aa6-a0a1-5cd07ac80426)

Any idea or suspect? Looks like it does not like whatever resolves into null or empty values in the actionGroupIds section.

Reply:
Issue fixed and code optimized ... pls continue with the review :)